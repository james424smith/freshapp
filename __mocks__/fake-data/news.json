{
  "pressReleases": [
    { "title": "some title", "imageUrl": "some blob" },
    { "title": "some title2", "imageUrl": "some blob2" },
    { "title": "some title3", "imageUrl": "some blob3" },
    { "title": "some title4", "imageUrl": "some blob4" }
  ],
  "newsletters": [
    {
      "document": "https://marlow-navigation.com/files/marlow_redaktion/downloads/newsletter_archiv/Marlow_Navigation_Newsletter_23rd_Issue_Semester_2_2018.pdf",
      "title": "23RD ISSUE - SEMESTER 2 / 2018"
    },
    {
      "document": "https://marlow-navigation.com/files/marlow_redaktion/downloads/newsletter_archiv/Marlow_Navigation_Newsletter_22nd_Issue_Semester_1_2018.pdf",
      "title": "22ND ISSUE - SEMESTER 1 / 2018"
    }
  ]
}
