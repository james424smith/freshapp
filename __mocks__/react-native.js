import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
