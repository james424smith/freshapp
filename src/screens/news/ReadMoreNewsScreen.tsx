import React from "react";
import { View } from "react-native";
import constants from "../../constants/config";
import WebRender from "../../components/WebRender";
import styles, { stylesheet } from "../../styles/more_screens/NewsCardStyles";
import { NEWSLETTERS } from "../../constants/NewsConstants";
import { AssignmentStackParamList } from "../../navigation/MainTabNavigator";
import { RouteProp } from "@react-navigation/native";
import PdfRender from "../../components/PdfRender";
import Colors from "../../constants/styles/colors";

interface Props {
  route: RouteProp<AssignmentStackParamList, "LatestReleases">;
}

export default (props: Props) => {
  const { route } = props;
  const imagePrefix = constants().IMAGES_PREFIX_URL;

  const newItem = route.params?.newItem;
  return newItem.newItemType === NEWSLETTERS ? (
    <PdfRender document={newItem.document} resourceType={"url"} />
  ) : (
    <View
      style={{
        ...styles.containerWebRender,
        backgroundColor: Colors.white,
      }}
    >
      <WebRender
        stringToRender={`<html>${stylesheet}<body><img <href src="${imagePrefix}/${newItem?.imageUrl}" />${newItem?.text}<body></html>`}
        resourceType={"html"}
      />
    </View>
  );
};
