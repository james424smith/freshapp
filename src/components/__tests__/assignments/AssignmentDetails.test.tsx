import React from "react";
import { render, cleanup } from "@testing-library/react-native";
import AssignmentDetails from "../../assignments/AssignmentDetails";
import { assignmentDetails } from "../../../../__mocks__/fake-data/assignmentDetails.json";
import news from "../../../../__mocks__/fake-data/news.json";

describe("AssignmentDetails test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render AssignmentDetails correctly as per the ui made when the employmentOffer is false and pressReleases are more than 3 ", () => {
    const { toJSON } = render(
      <AssignmentDetails
        availabilityDate={"01-01-2021"}
        initials={"GI"}
        imageUri={{ uri: "1234" }}
        assignmentType={"current"}
        handleAssignmentType={() => {}}
        assignmentDetails={assignmentDetails}
        news={news}
        hasEmployment={false}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it("should render AssignmentDetails correctly as per the ui made when the employmentOffer is false and pressReleases are less than 3 ", () => {
    const { toJSON } = render(
      <AssignmentDetails
        availabilityDate={"01-01-2021"}
        initials={"GI"}
        imageUri={{ uri: "1234" }}
        assignmentType={"current"}
        handleAssignmentType={() => {}}
        assignmentDetails={assignmentDetails}
        news={{ newsletters: news.newsletters, pressReleases: [] }}
        hasEmployment={false}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it("should render AssignmentDetails correctly as per the ui made when the employmentOffer is false and pressReleases  is undefined", () => {
    const { toJSON } = render(
      <AssignmentDetails
        availabilityDate={"01-01-2021"}
        initials={"GI"}
        imageUri={{ uri: "1234" }}
        assignmentType={"current"}
        handleAssignmentType={() => {}}
        assignmentDetails={assignmentDetails}
        news={{ newsletters: news.newsletters }}
        hasEmployment={false}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it("should render AssignmentDetails correctly as per the ui made when the employmentOffer is true", () => {
    const { toJSON } = render(
      <AssignmentDetails
        availabilityDate={"01-01-2021"}
        initials={"GI"}
        imageUri={{ uri: "1234" }}
        assignmentType={"current"}
        handleAssignmentType={() => {}}
        assignmentDetails={assignmentDetails}
        news={news}
        hasEmployment={true}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
