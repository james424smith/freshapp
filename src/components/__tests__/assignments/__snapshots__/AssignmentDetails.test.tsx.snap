// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssignmentDetails test should render AssignmentDetails correctly as per the ui made when the employmentOffer is false and pressReleases  is undefined 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginTop": 34,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": undefined,
          "flexGrow": 1,
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "color": undefined,
              "fontSize": 37.5,
              "fontWeight": "bold",
              "justifyContent": "center",
              "lineHeight": 53.5,
              "paddingBottom": 15,
              "paddingLeft": 22.5,
              "paddingRight": 15,
              "paddingTop": 15,
            },
          ]
        }
      >
        Latest Press Releases
      </Text>
      <RCTScrollView
        activeAnimationOptions={null}
        activeAnimationType="timing"
        activeSlideAlignment="center"
        activeSlideOffset={20}
        apparitionDelay={0}
        automaticallyAdjustContentInsets={false}
        autoplay={false}
        autoplayDelay={1000}
        autoplayInterval={3000}
        callbackOffsetMargin={5}
        containerCustomStyle={Object {}}
        contentContainerCustomStyle={Object {}}
        contentContainerStyle={
          Array [
            Object {
              "paddingLeft": -318.75,
              "paddingRight": -318.75,
            },
            Object {},
          ]
        }
        data={Array []}
        decelerationRate="fast"
        directionalLockEnabled={true}
        disableVirtualization={false}
        enableMomentum={false}
        enableSnap={true}
        firstItem={0}
        forwardedRef={[Function]}
        getItem={[Function]}
        getItemCount={[Function]}
        hasParallaxImages={false}
        horizontal={true}
        inactiveSlideOpacity={0.7}
        inactiveSlideScale={0.9}
        inactiveSlideShift={0}
        initialNumToRender={5}
        invertStickyHeaders={false}
        inverted={false}
        itemWidth={600}
        keyExtractor={[Function]}
        layout="default"
        lockScrollTimeoutDuration={1000}
        lockScrollWhileSnapping={false}
        loop={false}
        loopClonesPerSide={3}
        maxToRenderPerBatch={11}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onResponderRelease={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        overScrollMode="never"
        pinchGestureEnabled={false}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEnabled={true}
        scrollEventThrottle={1}
        scrollsToTop={false}
        shouldOptimizeUpdates={true}
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        slideStyle={Object {}}
        sliderWidth={-37.5}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "flexDirection": "row",
            "opacity": 0,
            "width": -37.5,
          }
        }
        swipeThreshold={20}
        updateCellsBatchingPeriod={50}
        useScrollView={false}
        vertical={false}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={11}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
</View>
`;

exports[`AssignmentDetails test should render AssignmentDetails correctly as per the ui made when the employmentOffer is false and pressReleases are less than 3  1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginTop": 34,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": undefined,
          "flexGrow": 1,
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "color": undefined,
              "fontSize": 37.5,
              "fontWeight": "bold",
              "justifyContent": "center",
              "lineHeight": 53.5,
              "paddingBottom": 15,
              "paddingLeft": 22.5,
              "paddingRight": 15,
              "paddingTop": 15,
            },
          ]
        }
      >
        Latest Press Releases
      </Text>
      <RCTScrollView
        activeAnimationOptions={null}
        activeAnimationType="timing"
        activeSlideAlignment="center"
        activeSlideOffset={20}
        apparitionDelay={0}
        automaticallyAdjustContentInsets={false}
        autoplay={false}
        autoplayDelay={1000}
        autoplayInterval={3000}
        callbackOffsetMargin={5}
        containerCustomStyle={Object {}}
        contentContainerCustomStyle={Object {}}
        contentContainerStyle={
          Array [
            Object {
              "paddingLeft": -318.75,
              "paddingRight": -318.75,
            },
            Object {},
          ]
        }
        data={Array []}
        decelerationRate="fast"
        directionalLockEnabled={true}
        disableVirtualization={false}
        enableMomentum={false}
        enableSnap={true}
        firstItem={0}
        forwardedRef={[Function]}
        getItem={[Function]}
        getItemCount={[Function]}
        hasParallaxImages={false}
        horizontal={true}
        inactiveSlideOpacity={0.7}
        inactiveSlideScale={0.9}
        inactiveSlideShift={0}
        initialNumToRender={5}
        invertStickyHeaders={false}
        inverted={false}
        itemWidth={600}
        keyExtractor={[Function]}
        layout="default"
        lockScrollTimeoutDuration={1000}
        lockScrollWhileSnapping={false}
        loop={false}
        loopClonesPerSide={3}
        maxToRenderPerBatch={11}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onResponderRelease={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        overScrollMode="never"
        pinchGestureEnabled={false}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEnabled={true}
        scrollEventThrottle={1}
        scrollsToTop={false}
        shouldOptimizeUpdates={true}
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        slideStyle={Object {}}
        sliderWidth={-37.5}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "flexDirection": "row",
            "opacity": 0,
            "width": -37.5,
          }
        }
        swipeThreshold={20}
        updateCellsBatchingPeriod={50}
        useScrollView={false}
        vertical={false}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={11}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
</View>
`;

exports[`AssignmentDetails test should render AssignmentDetails correctly as per the ui made when the employmentOffer is false and pressReleases are more than 3  1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginTop": 34,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": undefined,
          "flexGrow": 1,
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "color": undefined,
              "fontSize": 37.5,
              "fontWeight": "bold",
              "justifyContent": "center",
              "lineHeight": 53.5,
              "paddingBottom": 15,
              "paddingLeft": 22.5,
              "paddingRight": 15,
              "paddingTop": 15,
            },
          ]
        }
      >
        Latest Press Releases
      </Text>
      <RCTScrollView
        activeAnimationOptions={null}
        activeAnimationType="timing"
        activeSlideAlignment="center"
        activeSlideOffset={20}
        apparitionDelay={0}
        automaticallyAdjustContentInsets={false}
        autoplay={false}
        autoplayDelay={1000}
        autoplayInterval={3000}
        callbackOffsetMargin={5}
        containerCustomStyle={Object {}}
        contentContainerCustomStyle={Object {}}
        contentContainerStyle={
          Array [
            Object {
              "paddingLeft": -318.75,
              "paddingRight": -318.75,
            },
            Object {},
          ]
        }
        data={
          Array [
            Object {
              "imageUrl": "some blob",
              "title": "some title",
            },
            Object {
              "imageUrl": "some blob2",
              "title": "some title2",
            },
            Object {
              "imageUrl": "some blob3",
              "title": "some title3",
            },
          ]
        }
        decelerationRate="fast"
        directionalLockEnabled={true}
        disableVirtualization={false}
        enableMomentum={false}
        enableSnap={true}
        firstItem={0}
        forwardedRef={[Function]}
        getItem={[Function]}
        getItemCount={[Function]}
        hasParallaxImages={false}
        horizontal={true}
        inactiveSlideOpacity={0.7}
        inactiveSlideScale={0.9}
        inactiveSlideShift={0}
        initialNumToRender={5}
        invertStickyHeaders={false}
        inverted={false}
        itemWidth={600}
        keyExtractor={[Function]}
        layout="default"
        lockScrollTimeoutDuration={1000}
        lockScrollWhileSnapping={false}
        loop={false}
        loopClonesPerSide={3}
        maxToRenderPerBatch={11}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onResponderRelease={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        overScrollMode="never"
        pinchGestureEnabled={false}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEnabled={true}
        scrollEventThrottle={1}
        scrollsToTop={false}
        shouldOptimizeUpdates={true}
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        slideStyle={Object {}}
        sliderWidth={-37.5}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "flexDirection": "row",
            "opacity": 0,
            "width": -37.5,
          }
        }
        swipeThreshold={20}
        updateCellsBatchingPeriod={50}
        useScrollView={false}
        vertical={false}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={11}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "opacity": 1,
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                  "width": 600,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "rgb(0, 122, 255)",
                    "borderBottomLeftRadius": 5,
                    "borderBottomRightRadius": 5,
                    "borderTopLeftRadius": 5,
                    "borderTopRightRadius": 5,
                    "elevation": 5,
                    "marginBottom": 37.5,
                    "marginLeft": 22.5,
                    "paddingBottom": 22.5,
                    "paddingLeft": 22.5,
                    "shadowColor": "#2A2E43",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 0.5,
                    "shadowRadius": 10,
                  }
                }
                testID="card-button"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#2A2E43",
                        "fontFamily": "Roboto-Regular",
                      },
                      Object {
                        "color": "rgb(28, 28, 30)",
                        "fontSize": 29.5,
                        "fontWeight": "bold",
                        "marginTop": 24,
                        "minHeight": 73.5,
                        "paddingBottom": 7.5,
                        "paddingLeft": 15,
                        "paddingRight": 15,
                      },
                    ]
                  }
                >
                  some title
                </Text>
                <Text
                  numberOfLines={4}
                  style={
                    Array [
                      Object {
                        "color": "#2A2E43",
                        "fontFamily": "Roboto-Regular",
                      },
                      Object {
                        "color": "rgb(28, 28, 30)",
                        "fontSize": 26.5,
                        "paddingLeft": 15,
                        "paddingRight": 15,
                      },
                    ]
                  }
                />
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="read-more-card-button"
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#2A2E43",
                          "fontFamily": "Roboto-Regular",
                        },
                        Object {
                          "color": undefined,
                          "fontSize": 26.5,
                          "marginTop": 37.5,
                          "paddingRight": 30,
                          "textAlign": "right",
                        },
                      ]
                    }
                  >
                    Read More
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "opacity": 0.7,
                  "transform": Array [
                    Object {
                      "scale": 0.9,
                    },
                  ],
                  "width": 600,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "rgb(0, 122, 255)",
                    "borderBottomLeftRadius": 5,
                    "borderBottomRightRadius": 5,
                    "borderTopLeftRadius": 5,
                    "borderTopRightRadius": 5,
                    "elevation": 5,
                    "marginBottom": 37.5,
                    "marginLeft": 22.5,
                    "paddingBottom": 22.5,
                    "paddingLeft": 22.5,
                    "shadowColor": "#2A2E43",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 0.5,
                    "shadowRadius": 10,
                  }
                }
                testID="card-button"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#2A2E43",
                        "fontFamily": "Roboto-Regular",
                      },
                      Object {
                        "color": "rgb(28, 28, 30)",
                        "fontSize": 29.5,
                        "fontWeight": "bold",
                        "marginTop": 24,
                        "minHeight": 73.5,
                        "paddingBottom": 7.5,
                        "paddingLeft": 15,
                        "paddingRight": 15,
                      },
                    ]
                  }
                >
                  some title2
                </Text>
                <Text
                  numberOfLines={4}
                  style={
                    Array [
                      Object {
                        "color": "#2A2E43",
                        "fontFamily": "Roboto-Regular",
                      },
                      Object {
                        "color": "rgb(28, 28, 30)",
                        "fontSize": 26.5,
                        "paddingLeft": 15,
                        "paddingRight": 15,
                      },
                    ]
                  }
                />
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="read-more-card-button"
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#2A2E43",
                          "fontFamily": "Roboto-Regular",
                        },
                        Object {
                          "color": undefined,
                          "fontSize": 26.5,
                          "marginTop": 37.5,
                          "paddingRight": 30,
                          "textAlign": "right",
                        },
                      ]
                    }
                  >
                    Read More
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "opacity": 0.7,
                  "transform": Array [
                    Object {
                      "scale": 0.9,
                    },
                  ],
                  "width": 600,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "rgb(0, 122, 255)",
                    "borderBottomLeftRadius": 5,
                    "borderBottomRightRadius": 5,
                    "borderTopLeftRadius": 5,
                    "borderTopRightRadius": 5,
                    "elevation": 5,
                    "marginBottom": 37.5,
                    "marginLeft": 22.5,
                    "paddingBottom": 22.5,
                    "paddingLeft": 22.5,
                    "shadowColor": "#2A2E43",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 0.5,
                    "shadowRadius": 10,
                  }
                }
                testID="card-button"
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#2A2E43",
                        "fontFamily": "Roboto-Regular",
                      },
                      Object {
                        "color": "rgb(28, 28, 30)",
                        "fontSize": 29.5,
                        "fontWeight": "bold",
                        "marginTop": 24,
                        "minHeight": 73.5,
                        "paddingBottom": 7.5,
                        "paddingLeft": 15,
                        "paddingRight": 15,
                      },
                    ]
                  }
                >
                  some title3
                </Text>
                <Text
                  numberOfLines={4}
                  style={
                    Array [
                      Object {
                        "color": "#2A2E43",
                        "fontFamily": "Roboto-Regular",
                      },
                      Object {
                        "color": "rgb(28, 28, 30)",
                        "fontSize": 26.5,
                        "paddingLeft": 15,
                        "paddingRight": 15,
                      },
                    ]
                  }
                />
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="read-more-card-button"
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#2A2E43",
                          "fontFamily": "Roboto-Regular",
                        },
                        Object {
                          "color": undefined,
                          "fontSize": 26.5,
                          "marginTop": 37.5,
                          "paddingRight": 30,
                          "textAlign": "right",
                        },
                      ]
                    }
                  >
                    Read More
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;

exports[`AssignmentDetails test should render AssignmentDetails correctly as per the ui made when the employmentOffer is true 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginTop": 34,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": undefined,
          "borderRadius": 16,
          "borderWidth": 1,
          "flexBasis": "45%",
          "height": 160,
          "justifyContent": "center",
          "margin": 15,
          "marginTop": 30,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "alignItems": "center",
              "color": undefined,
              "fontSize": 24,
              "justifyContent": "center",
            },
          ]
        }
      >
        Joining Date
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "alignItems": "center",
              "color": undefined,
              "fontSize": 37.5,
              "fontWeight": "bold",
              "justifyContent": "center",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": undefined,
          "borderRadius": 16,
          "borderWidth": 1,
          "flexBasis": "45%",
          "height": 160,
          "justifyContent": "center",
          "margin": 15,
          "marginTop": 30,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "alignItems": "center",
              "color": undefined,
              "fontSize": 24,
              "justifyContent": "center",
            },
          ]
        }
      >
        EOC Date
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "alignItems": "center",
              "color": undefined,
              "fontSize": 37.5,
              "fontWeight": "bold",
              "justifyContent": "center",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(0, 122, 255)",
          "borderColor": undefined,
          "borderRadius": 16,
          "borderWidth": 1,
          "flexBasis": "45%",
          "height": 160,
          "justifyContent": "center",
          "margin": 15,
          "marginBottom": 15,
          "marginTop": 15,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "alignItems": "center",
              "color": undefined,
              "fontSize": 24,
              "justifyContent": "center",
            },
          ]
        }
      >
        Port (tentative)
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "alignItems": "center",
              "color": "rgb(28, 28, 30)",
              "fontSize": 26.5,
              "fontWeight": "bold",
              "justifyContent": "center",
              "textAlign": "center",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(0, 122, 255)",
          "borderColor": undefined,
          "borderRadius": 16,
          "borderWidth": 1,
          "flexBasis": "45%",
          "height": 160,
          "justifyContent": "center",
          "margin": 15,
          "marginBottom": 15,
          "marginTop": 15,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "alignItems": "center",
              "color": undefined,
              "fontSize": 24,
              "justifyContent": "center",
            },
          ]
        }
      >
        Vessel Name
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "alignItems": "center",
              "color": "rgb(28, 28, 30)",
              "fontSize": 26.5,
              "fontWeight": "bold",
              "justifyContent": "center",
              "textAlign": "center",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(0, 122, 255)",
          "borderColor": undefined,
          "borderRadius": 16,
          "borderWidth": 1,
          "flexBasis": "45%",
          "height": 160,
          "justifyContent": "center",
          "margin": 15,
          "marginBottom": 15,
          "marginTop": 15,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "alignItems": "center",
              "color": undefined,
              "fontSize": 24,
              "justifyContent": "center",
            },
          ]
        }
      >
        Vessel Flag
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "alignItems": "center",
              "color": "rgb(28, 28, 30)",
              "fontSize": 26.5,
              "fontWeight": "bold",
              "justifyContent": "center",
              "textAlign": "center",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(0, 122, 255)",
          "borderColor": undefined,
          "borderRadius": 16,
          "borderWidth": 1,
          "flexBasis": "45%",
          "height": 160,
          "justifyContent": "center",
          "margin": 15,
          "marginBottom": 15,
          "marginTop": 15,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "alignItems": "center",
              "color": undefined,
              "fontSize": 24,
              "justifyContent": "center",
            },
          ]
        }
      >
        Vessel IMO No.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "alignItems": "center",
              "color": "rgb(28, 28, 30)",
              "fontSize": 26.5,
              "fontWeight": "bold",
              "justifyContent": "center",
              "textAlign": "center",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
